src/mlpack/methods/kmeans/kill_empty_clusters.hpp

```diff
@@ -55,9 +55,12 @@ class KillEmptyClusters
       MetricType& /* metric */,
       const size_t /* iteration */)
   {
-    // Kill the empty cluster.
-    newCentroids.col(emptyCluster).fill(DBL_MAX);
-    return 0; // No points were changed.
+    // Remove the empty cluster.
+      if (emptyCluster < newCentroids.n_cols)
+      {
+          newCentroids.shed_col(emptyCluster);
+          return 0; // No points were changed.
```

Comment:
Don't we have to update `counts` also?

Reply:
I don't think we need to update the `count` as `count` represent the number of points in a particular cluster and dead cluster will not have any points in it. If we really want to modify the `counts` then we have to delete `count[i]` ( here `i` is the index of dead cluster) and then we also have to modify the `cluster` i.e. `cluster-=1` . I can do whichever way you want me to do :)

Reply:
I think we do have to update `counts`, but if you write a test we can know for sure if it is needed.

Reply:
I've updated `counts`  as well as written the test :)